main.rb

```diff
@@ -1,45 +1,21 @@
 require './app'
+require './option'
+require './show_menu'
 
 class Main
   def initialize
     @app = App.new
   end
 
-  def menu
-    puts 'Welcome to School Library App!'
+  def main
+    puts 'welcome to school library App!'
     loop do
-      puts 'Please choose by using one of the following numbers!'
-      puts '1 - List all books'
-      puts '2 - List all people'
-      puts '3 - Create a person'
-      puts '4 - Create a book'
-      puts '5 - Create a rental'
-      puts '6 - List all rentals for a given person id'
-      puts '7 - Exit'
-      option = gets.chomp
-
-      case option
-      when '1'
-        @app.list_all_books
-      when '2'
-        @app.list_all_people
-      when '3'
-        @app.create_person
-      when '4'
-        @app.create_book
-      when '5'
-        @app.create_rental
-      when '6'
-        @app.list_rentals
-      when '7'
-        puts 'Thank you for using our library ¯\^-^/¯'
-        return
-      else
-        puts 'Please enter a number between 1 and 7'
-      end
+      show_menu
```

Comment:
* I can see in your `App.rb` file some methods have not been refactored to comply with  the "Single-responsibility Principle". eg: the create student method should be responsible for only creating students below 👇 are good examples of Single responsability rule being followed
````ruby
def list_books
    @books.each do |book|
      puts book
    end
  end
````
````ruby
 def list_people
    @people.each do |person|
      puts person
    end
  end
````
* Kindly [follow this lesson for a reminder on how to refactor your code properly](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/solid.md#best-practices-and-solid) and ensure `all requests for input and input validation should've been placed in the main.rb file and/or classes/methods whose responsibilities are to request or validate input`

Reply:
```
def create_student
    print 'Name:'
    name = gets.chomp

    print 'Age:'
    age = gets.chomp.to_i

    print 'Has Parent permission? [Y/N]:'
    parent_permission = gets.chomp.downcase

    student = Student.new(age, @classroom, name, parent_permission: parent_permission == 'y')
    @people << student
    puts 'Student created successfully'
  end
```
Bundle of thanks❤ dear @Davidosky007,
As per the above code in my `app.rb ` file, the create_student method is responsible for just creating students nothing else, and the create_teacher method is as well, so how I can refactor it more than that?
It would be great to give a little bit more description in mentioned part.
once again thank you very much👏👏👏